proxy/session.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type session struct {
+	IsActive    bool
+	MediatorURL string `json:"mediatorUrl"`
+}
+
+func sessionHandler(currentSession *session) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/session" {
+			return
+		}
+		if req.Method == "POST" {
+			startSession(currentSession, req)
+		}
+		if req.Method == "DELETE" {
+			currentSession.IsActive = false
```

Comment:
To avoid issues in the future with properties that are not cleaned here it will be best if we just create new instance here rather than reusing the existing one.